src/Raven.Client/Documents/Operations/Revisions/DeleteRevisionsOperation.cs

```diff
@@ -0,0 +1,285 @@
+        {
+            var parameters = new Parameters()
+            {
+                DocumentIds = new List<string> { documentId },
+                RevisionsChangeVectors = revisionsChangeVectors,
+                RemoveForceCreatedRevisions = removeForceCreatedRevisions
+            };
+            parameters.Validate();
+            _parameters = parameters;
+        }
+
+        public RavenCommand<Result> GetCommand(DocumentConventions conventions, JsonOperationContext context)
+        {
+            return new DeleteRevisionsCommand(conventions, _parameters);
+        }
+
+        internal sealed class Parameters : ICloneable
+        {
+            public List<string> DocumentIds { get; set; }
+
+            public bool RemoveForceCreatedRevisions { get; set; }
+
+            // Either!
+            public List<string> RevisionsChangeVectors { get; set; }
+
+            // Or!
+            public DateTime? From { get; set; } // start
+            public DateTime? To { get; set; } // end
+
+            internal void Validate()
+            {
+                if (DocumentIds.IsNullOrEmpty())
+                    throw new ArgumentNullException(nameof(DocumentIds), $"request '{nameof(DocumentIds)}' cannot be null or empty.");
+
+                foreach (var id in DocumentIds)
+                {
+                    if (string.IsNullOrEmpty(id))
+                        throw new ArgumentNullException(nameof(DocumentIds), $"request '{nameof(DocumentIds)}' contains null or empty ids.");
+                }
+
+                if (RevisionsChangeVectors.IsNullOrEmpty())
+                {
+                    if (From.HasValue && To.HasValue && From >= To)
+                        throw new ArgumentException($"{nameof(From)}, {nameof(To)}", "'After' must be greater then 'Before'.");
+                }
+                else
+                {
+                    if (DocumentIds.Count > 1)
+                        throw new ArgumentException(nameof(DocumentIds), "The request must include exactly one document ID when deleting specific revisions by their change-vectors.");
+
+                    if (From.HasValue)
+                        throw new ArgumentException(nameof(From), $"request '{nameof(From)}' cannot have a value when deleting specific revisions by their change-vectors.");
+
+                    if (To.HasValue)
+                        throw new ArgumentException(nameof(To), $"request '{nameof(To)}' cannot have a value when deleting specific revisions by their change-vectors.");
+
+                    foreach (var cv in RevisionsChangeVectors)
+                    {
+                        if (string.IsNullOrEmpty(cv))
+                            throw new ArgumentNullException(nameof(RevisionsChangeVectors), $"request '{nameof(RevisionsChangeVectors)}' contains null or empty change-vectors.");
+                    }
+                }
+
+            }
+
+            public object Clone()
```

Comment:
remove `ICloneable` and pass `DocumentIds`

Reply:
Done